node/src/actors/chain_manager/mod.rs

```diff
@@ -1251,46 +1251,53 @@ impl ChainManager {
 
                         let chain_info = act.chain_state.chain_info.as_ref().unwrap();
                         let reputation_engine = act.chain_state.reputation_engine.as_ref().unwrap();
+                        let bootstrap_committee = consensus_constants.get_bootstrap_committee();
 
-                        let reputed_ars_members =
+                        let (reputed_ars_members, committee_size) =
                             // Before reaching the epoch activity_period + collateral_age the bootstrap committee signs the superblock
                             // collateral_age is measured in blocks instead of epochs, but this only means that the period in which
                             // the bootstrap committee signs is at least epoch activity_period + collateral_age
                             if block_epoch
                                 > chain_info.consensus_constants.collateral_age
                                     + chain_info.consensus_constants.activity_period
                             {
-                                let ars_members = reputation_engine.get_rep_ordered_ars_list();
-                                let reputed = reputed_ars(&ars_members, &reputation_engine);
-
-                                // In case of no reputed nodes, return all active nodes
-                                if reputed.is_empty() {
-                                    ars_members
-                                } else {
-                                    reputed
-                                }
-                            } else {
-                                chain_info
-                                    .consensus_constants
-                                    .bootstrapping_committee
-                                    .iter()
-                                    .map(|add| add.parse().expect("Malformed bootstrapping committee"))
-                                    .collect()
+                                let ars_members = {
+                                    let ars_list = reputation_engine.get_rep_ordered_ars_list();
+                                    let reputed = reputed_ars(&ars_list, &reputation_engine);
+                                    // In case of no reputed nodes, return all active nodes
+                                    if reputed.is_empty() {
+                                        ars_list
+                                    } else {
+                                        reputed
+                                    }
+                                };
+                                let committee_size = {
+                                    if act.chain_state.superblock_state.get_committee() == bootstrap_committee.into_iter().collect::<HashSet<PublicKeyHash>>() {
```

Comment:
This checks if the last superblock has been signed by the bootstrap committee. But wouldn't it be easier to just check the superblock index, and compare it to a variable like `last_superblock_index_that_will_be_signed_by_the_bootstrap_committee`?

Reply:
definitely easier, I can modify that